### PR TITLE
OCM-10638 | test: Automated ids:54864,73672,66761

### PR DESCRIPTION
**Pull request by yingzhanredhat**
https://privatebin.corp.redhat.com/?cab366c9fd23df41#AK7z6bLxqoSnQASRia3fQM4PPz756bBQuDrPFKP5jk2K

### CODE REVIEW

File: tests/utils/exec/rosacli/version_service.go [hunk 1]
```diff
@@ -122,14 +126,15 @@ func (v *versionService) ListAndReflectVersions(
 	return versionList, err
 }
 func (v *versionService) ListAndReflectJsonVersions(
-	channelGroup string, hostedCP bool, flags ...string) (versionList *OpenShiftVersionTableList, err error) {
+	channelGroup string, hostedCP bool, flags ...string) (versionList []*OpenShiftVersionJsonOutput, err error) {
 	var output bytes.Buffer
+	flags = append(flags, "-ojson")
```
Comment (yuwang-RH) [inline tests/utils/exec/rosacli/version_service.go:131]: An empty space between '-o' and 'json'?

Comment (yingzhanredhat) [reply tests/utils/exec/rosacli/version_service.go:131]: @yuwang-RH It works well for "-ojson". Is it required to add space?

File: tests/e2e/test_rosacli_version.go [hunk 2]
```diff
@@ -53,4 +54,46 @@ var _ = Describe("Get CLI version",
...
+						"INFO: Hosted cluster upgrades are cluster-based. To list available upgrades for a cluster, please use 'rosa list upgrades'"))
+
+				By("list the versions with channel group")
+				channelGroups := []string{"stable", "", "candidate", "nightly"}
+				for _, c := range channelGroups {
+					verList, err := versionService.ListAndReflectJsonVersions(c, true)
+					Expect(err).ToNot(HaveOccurred())
+					for _, v := range verList {
+						if !v.Enabled {
+							continue
```
Comment (yuwang-RH) [inline tests/e2e/test_rosacli_version.go:80]: Could you confirm if the steps bellow this line in the for loop can be checked? It seems all the below steps are skipped. The v.Enabled is always false as struct OpenShiftVersionTableOutput has no 'Enabled' field

Comment (yingzhanredhat) [reply tests/e2e/test_rosacli_version.go:80]: I added one print character under the if condition. The below steps are run. v.Enabled is to check the enabled/vaild version. And v struct is OpenShiftVersionJsonOutput.

File: tests/utils/exec/rosacli/version_service.go (hunk 1 above)
Comment (yuwang-RH) [reply tests/utils/exec/rosacli/version_service.go:131]: If it works as well, that's fine for me

File: tests/e2e/test_rosacli_version.go (hunk 2 above)
Comment (yuwang-RH) [reply tests/e2e/test_rosacli_version.go:81]: Ok, that's a change included in this PR. sorry to miss that

File: tests/e2e/test_rosacli_version.go [hunk 3]
```diff
@@ -53,4 +54,47 @@ var _ = Describe("Get CLI version",
...
+				versionService := rosaClient.Version
+
+				output, err := versionService.ListVersions("", true)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(rosaClient.Parser.TextData.Input(output).Parse().Tip()).
+					Should(ContainSubstring(
+						"INFO: Hosted cluster upgrades are cluster-based. " +
+							"To list available upgrades for a cluster, please use 'rosa list upgrades'"))
+
+				By("list the versions with channel group")
```
Comment (yuwang-RH) [inline tests/e2e/test_rosacli_version.go:74]: If it uses ListAndReflectJsonVersions to check the version, how can it ensure the output `rosa list version --hosted-cp` without '-o json' is correct, this command with the "-o" flag is more often used by customers

**Pull request merged**